src/um/webauthn.rs

```diff
@@ -0,0 +1,221 @@
+pub const WEBAUTHN_HASH_ALGORITHM_SHA_512: &'static str = "SHA-512";
+pub const WEBAUTHN_RP_ENTITY_INFORMATION_CURRENT_VERSION: DWORD = 1;
+STRUCT!{struct WEBAUTHN_RP_ENTITY_INFORMATION {
+    dwVersion: DWORD,
+    pwszId: PCWSTR,
+    pwszName: PCWSTR,
+    pwszIcon: PCWSTR,
+}}
+pub type PWEBAUTHN_RP_ENTITY_INFORMATION = *mut WEBAUTHN_RP_ENTITY_INFORMATION;
+pub type PCWEBAUTHN_RP_ENTITY_INFORMATION = *const WEBAUTHN_RP_ENTITY_INFORMATION;
+pub const WEBAUTHN_MAX_USER_ID_LENGTH: DWORD = 64;
+pub const WEBAUTHN_USER_ENTITY_INFORMATION_CURRENT_VERSION: DWORD = 1;
+STRUCT!{struct WEBAUTHN_USER_ENTITY_INFORMATION {
+    dwVersion: DWORD,
+    cbId: DWORD,
+    pbId: PBYTE,
+    pwszName: PCWSTR,
+    pwszIcon: PCWSTR,
+    pwszDisplayName: PCWSTR,
+}}
+pub type PWEBAUTHN_USER_ENTITY_INFORMATION = *mut WEBAUTHN_USER_ENTITY_INFORMATION;
+pub type PCWEBAUTHN_USER_ENTITY_INFORMATION = *const WEBAUTHN_USER_ENTITY_INFORMATION;
+pub const WEBAUTHN_CLIENT_DATA_CURRENT_VERSION: DWORD = 1;
+STRUCT!{struct WEBAUTHN_CLIENT_DATA {
+    dwVersion: DWORD,
+    cbClientDataJSON: DWORD,
+    pbClientDataJSON: PBYTE,
+    pwszHashAlgId: LPCWSTR,
+}}
+pub type PWEBAUTHN_CLIENT_DATA = *mut WEBAUTHN_CLIENT_DATA;
+pub type PCWEBAUTHN_CLIENT_DATA = *const WEBAUTHN_CLIENT_DATA;
+pub const WEBAUTHN_CREDENTIAL_TYPE_PUBLIC_KEY: &'static str = "public-key";
+pub const WEBAUTHN_COSE_CREDENTIAL_PARAMETER_CURRENT_VERSION: DWORD = 1;
+STRUCT!{struct WEBAUTHN_COSE_CREDENTIAL_PARAMETER {
+    dwVersion: DWORD,
+    pwszCredentialType: LPCWSTR,
+    lAlg: LONG,
+}}
+pub type PWEBAUTHN_COSE_CREDENTIAL_PARAMETER = *mut WEBAUTHN_COSE_CREDENTIAL_PARAMETER;
+pub type PCWEBAUTHN_COSE_CREDENTIAL_PARAMETER = *const WEBAUTHN_COSE_CREDENTIAL_PARAMETER;
+STRUCT!{struct WEBAUTHN_COSE_CREDENTIAL_PARAMETERS {
+    cCredentialParameters: DWORD,
+    pCredentialParameters: PWEBAUTHN_COSE_CREDENTIAL_PARAMETER,
+}}
+pub type PWEBAUTHN_COSE_CREDENTIAL_PARAMETERS = *mut WEBAUTHN_COSE_CREDENTIAL_PARAMETERS;
+pub type PCWEBAUTHN_COSE_CREDENTIAL_PARAMETERS = *const WEBAUTHN_COSE_CREDENTIAL_PARAMETERS;
+pub const WEBAUTHN_CREDENTIAL_CURRENT_VERSION: DWORD = 1;
+STRUCT!{struct WEBAUTHN_CREDENTIAL {
+    dwVersion: DWORD,
+    cbId: DWORD,
+    pbId: PBYTE,
+    pwszCredentialType: LPCWSTR,
+}}
+pub type PWEBAUTHN_CREDENTIAL = *mut WEBAUTHN_CREDENTIAL;
+pub type PCWEBAUTHN_CREDENTIAL = *const WEBAUTHN_CREDENTIAL;
+STRUCT!{struct WEBAUTHN_CREDENTIALS {
+    cCredentials: DWORD,
+    pCredentials: PWEBAUTHN_CREDENTIAL,
+}}
+pub type PWEBAUTHN_CREDENTIALS = *mut WEBAUTHN_CREDENTIALS;
+pub type PCWEBAUTHN_CREDENTIALS = *const WEBAUTHN_CREDENTIALS;
+pub const WEBAUTHN_CTAP_TRANSPORT_USB: DWORD = 0x01;
+pub const WEBAUTHN_CTAP_TRANSPORT_NFC: DWORD = 0x02;
+pub const WEBAUTHN_CTAP_TRANSPORT_BLE: DWORD = 0x04;
+pub const WEBAUTHN_CTAP_TRANSPORT_TEST: DWORD = 0x08;
+pub const WEBAUTHN_CTAP_TRANSPORT_INTERNAL: DWORD = 0x10;
+pub const WEBAUTHN_CTAP_TRANSPORT_FLAGS_MASK: DWORD = 0x1F;
```

Comment:
Should these be 0-padded like they are in the header ala GUIDs?  (values are correct)

Reply:
Changed in fc4d782e4118431d2e2517237f0b3d8704f70e38.